test_collections/matter/sdk_tests/support/python_testing/list_python_tests_classes.py

```diff
@@ -0,0 +1,147 @@
+# limitations under the License.
+#
+
+import ast
+import asyncio
+import json
+from pathlib import Path
+
+from test_collections.matter.config import matter_settings
+from test_collections.matter.sdk_tests.support.models.sdk_test_folder import (
+    SDKTestFolder,
+)
+from test_collections.matter.sdk_tests.support.sdk_container import SDKContainer
+
+SDK_TESTS_PATH = Path(__file__).parent.parent.parent
+PYTHON_TESTING_PATH = SDK_TESTS_PATH / "sdk_checkout/python_testing"
+JSON_OUTPUT_FILE_PATH = PYTHON_TESTING_PATH / "data.json"
+COMPLETE_JSON_OUTPUT_FILE_FOLDER = SDK_TESTS_PATH
+PYTHON_SCRIPTS_PATH = PYTHON_TESTING_PATH / "scripts/sdk"
+PYTHON_SCRIPTS_FOLDER = SDKTestFolder(path=PYTHON_SCRIPTS_PATH, filename_pattern="TC*")
+
+CONTAINER_GET_TEST_INFO_COMMAND = "--get_test_info"
+CONTAINER_TH_CLIENT_EXEC = (
+    "python3 /root/python_testing/scripts/sdk/test_harness_client.py"
+)
+
+
+def base_test_classes(module: ast.Module) -> list[ast.ClassDef]:
+    """Find classes that inherit from MatterBaseTest.
+
+    Args:
+        module (ast.Module): Python module.
+
+    Returns:
+        list[ast.ClassDef]: List of classes from the given module that inherit from
+        MatterBaseTest.
+    """
+    return [
+        c
+        for c in module.body
+        if isinstance(c, ast.ClassDef)
+        and any(
+            b for b in c.bases if isinstance(b, ast.Name) and b.id == "MatterBaseTest"
+        )
+    ]
+
+
+def get_command_list() -> list:
+    python_script_commands = []
+    python_test_files = PYTHON_SCRIPTS_FOLDER.file_paths(extension=".py")
+
+    for python_test_file in python_test_files:
+        parent_folder = python_test_file.parent.name
+        with open(python_test_file, "r") as python_file:
+            parsed_python_file = ast.parse(python_file.read())
+
+        test_classes = base_test_classes(parsed_python_file)
+        for test_class in test_classes:
+            script_command = [f"{parent_folder}/{python_test_file.stem}"]
+            script_command.append(f"{test_class.name}")
+            script_command.append(CONTAINER_GET_TEST_INFO_COMMAND)
+            python_script_commands.append(script_command)
+
+    return python_script_commands
+
+
+async def proccess_commands_sdk_container(commands: list) -> None:
+    complete_json = []
+    errors_found: list[str] = []
+    test_function_count = 0
+    invalid_test_function_count = 0
+
+    sdk_container: SDKContainer = SDKContainer()
+
+    await sdk_container.start()
+
+    for command in commands:
+        command_string = " ".join(command)
+        result = sdk_container.send_command(
+            command_string,
+            prefix=CONTAINER_TH_CLIENT_EXEC,
+        )
+        if result.exit_code != 0:
+            errors_found.append(f"Failed running command: {command}")
+            continue
+
+        with open(JSON_OUTPUT_FILE_PATH, "r") as json_file:
+            json_data = json.load(json_file)
+            json_file.close()
```

Comment:
```suggestion
            
```
I think when using `with` statement, it not necessary to close the file. Please check this information.

Reply:
You're correct Romulo, it's not necessary to close.
It's the purpose of the `with` statement indeed.